### PR TITLE
Finish reward actor with HAMT/AMT, params objects, cbor, exports

### PR DESCRIPTION
**Pull request by anorth**

### CODE REVIEW

File: actors/builtin/reward/reward_actor_state.go [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+	StartEpoch      abi.ChainEpoch
+	EndEpoch        abi.ChainEpoch
+	Value           abi.TokenAmount
+	AmountWithdrawn abi.TokenAmount
+}
+
+type VestingFunction int64
+const (
+	None VestingFunction = iota
+	Linear
```
Comment (zixuanzh) [inline actors/builtin/reward/reward_actor_state.go]: making a note that we might have one more option, more details to come next week cc @jzimmerman

File: actors/builtin/reward/reward_actor_state.go [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+)
+
+type InitialReward struct {
```
Comment (zixuanzh) [inline actors/builtin/reward/reward_actor_state.go]: What are these rewards? Investor tokens should already be in multisig.

File: gen/gen.go [hunk 3]
```diff
@@ -38,6 +39,19 @@ func main() {
...
 
+	if err := gen.WriteTupleEncodersToFile("./actors/builtin/reward/cbor_gen.go", "reward",
+		// actor state
+		reward.RewardActorState{},
+		// method params
+		reward.ConstructorParams{},
+		reward.AwardBlockRewardParams{},
+		// other types
+		reward.Reward{},
+		reward.InitialReward{},
```
Comment (zixuanzh) [inline gen/gen.go]: ditto

File: actors/builtin/reward/reward_actor_state.go (hunk 2 above)
Comment (anorth) [reply actors/builtin/reward/reward_actor_state.go]: Great! I'll remove this.

**Pull request merged**